From edb79e013dde121a82048157e1a7c66acdd30bed Mon Sep 17 00:00:00 2001
From: Pedro Palazon Candel <pedro@joyent.com>
Date: Mon, 17 Sep 2018 19:02:56 +0200
Subject: [PATCH] TRITON-770 Fix refreservation tests for kvm broken by
 TRITON-591 Reviewed by: Dave Eddy <dave.eddy@joyent.com> Reviewed by: Mike
 Gerdts <mike.gerdts@joyent.com>

---
 src/vm/node_modules/VM.js       | 19 ++++---------------
 src/vm/tests/test-create-kvm.js |  6 +++---
 2 files changed, 7 insertions(+), 18 deletions(-)

diff --git a/src/vm/node_modules/VM.js b/src/vm/node_modules/VM.js
index f7b97338..55bf7eea 100644
--- a/src/vm/node_modules/VM.js
+++ b/src/vm/node_modules/VM.js
@@ -1336,21 +1336,10 @@ exports.validate = function (brand, action, payload, options, callback)
                 var d_idx = 0;
                 if (payload.hasOwnProperty(d)) {
                     payload[d].forEach(function (disk) {
-                        if (disk.hasOwnProperty('refreservation')) {
-                            if (disk.refreservation < 0) {
-                                errors.bad_values.push(d + '.' + d_idx
-                                    + '.refreservation');
-                            } else if (disk.size
-                                && disk.refreservation > disk.size) {
-
-                                errors.bad_values.push(d + '.' + d_idx
-                                    + '.refreservation');
-                            } else if (disk.image_size
-                                && disk.refreservation > disk.image_size) {
-
-                                errors.bad_values.push(d + '.' + d_idx
-                                    + '.refreservation');
-                            }
+                        if (disk.hasOwnProperty('refreservation') &&
+                            disk.refreservation < 0) {
+                                errors.bad_values.push(d + '.' + d_idx +
+                                    '.refreservation');
                         }
                         d_idx++;
                     });
diff --git a/src/vm/tests/test-create-kvm.js b/src/vm/tests/test-create-kvm.js
index a9724c1c..f84c9edb 100644
--- a/src/vm/tests/test-create-kvm.js
+++ b/src/vm/tests/test-create-kvm.js
@@ -1,4 +1,4 @@
-// Copyright 2015 Joyent, Inc.  All rights reserved.
+// Copyright 2018 Joyent, Inc.  All rights reserved.
 //
 // These tests ensure that create works with specific options set.
 //
@@ -295,8 +295,8 @@ test('test default refreservation', function(t) {
                 }
                 disks = obj.disks;
                 t.ok(disks.length === 2, 'VM has 2 disks');
-                t.ok(disks[0].refreservation === disks[0].size, 'disk 0 has correct refreservation: ' + disks[0].refreservation + '/' + disks[0].size);
-                t.ok(disks[1].refreservation === disks[1].size, 'disk 1 has correct refreservation: ' + disks[1].refreservation + '/' + disks[1].size);
+                t.ok(disks[0].refreservation > disks[0].size, 'disk 0 has correct refreservation: ' + disks[0].refreservation + '>' + disks[0].size);
+                t.ok(disks[1].refreservation > disks[1].size, 'disk 1 has correct refreservation: ' + disks[1].refreservation + '>' + disks[1].size);
                 state.vmobj = obj;
                 cb();
             });
-- 
2.21.0

